python/pyarrow/table.pxi

```diff
@@ -1669,26 +1669,44 @@ def table(data, names=None, schema=None, metadata=None):
             "Expected pandas DataFrame, python dictionary or list of arrays")
 
 
-def concat_tables(tables):
+def concat_tables(tables, c_bool promote=False, MemoryPool memory_pool=None):
     """
-    Perform zero-copy concatenation of pyarrow.Table objects. Raises exception
-    if all of the Table schemas are not the same
+    Perform zero-copy concatenation of pyarrow.Table objects.
+
+    If promote==False, the schemas of all the Tables must be the same
+    (except the metadata), otherwise an exception will be raised. The result
+    Table will share the metadata with the first table.
+
+    If promote==True, columns of the same name will be concatenated.
```

Comment:
I don't think "columns of the same name will be concatenated" adds any information (since this is true in the False case, too). How about `If promote==True, any null type arrays will be casted to the type of other arrays in the column of the same name. If a"